Installing sassc broke on jruby-head on Windows

See https://github.com/ruby/setup-ruby/actions/runs/13279868644/job/37076053011?pr=705
It also fails on windows-2022 and windows-2025, so all jruby-head + windows.
It looks like a regression because AFAIK this is the first time it fails.

Output:
```
gem install sassc
Building native extensions. This could take a while...
ERROR:  Error installing sassc:
	ERROR: Failed to build gem native extension.

    current directory: D:/jruby-head/lib/ruby/gems/shared/gems/sassc-2.4.0/ext
D:/jruby-head/bin/jruby.exe extconf.rb
creating Makefile

current directory: D:/jruby-head/lib/ruby/gems/shared/gems/sassc-2.4.0/ext
make DESTDIR\= sitearchdir\=./.gem.20250212-744-b2t36v sitelibdir\=./.gem.20250212-744-b2t36v clean

current directory: D:/jruby-head/lib/ruby/gems/shared/gems/sassc-2.4.0/ext
make DESTDIR\= sitearchdir\=./.gem.20250212-744-b2t36v sitelibdir\=./.gem.20250212-744-b2t36v
compiling ./libsass/src/ast.cpp
compiling ./libsass/src/ast2c.cpp
compiling ./libsass/src/ast_fwd_decl.cpp
compiling ./libsass/src/ast_sel_cmp.cpp
compiling ./libsass/src/ast_sel_super.cpp
compiling ./libsass/src/ast_sel_unify.cpp
compiling ./libsass/src/ast_sel_weave.cpp
compiling ./libsass/src/ast_selectors.cpp
compiling ./libsass/src/ast_supports.cpp
compiling ./libsass/src/ast_values.cpp
compiling ./libsass/src/backtrace.cpp
compiling ./libsass/src/base64vlq.cpp
compiling ./libsass/src/bind.cpp
compiling ./libsass/src/c2ast.cpp
compiling ./libsass/src/c99func.c
compiling ./libsass/src/cencode.c
compiling ./libsass/src/check_nesting.cpp
compiling ./libsass/src/color_maps.cpp
compiling ./libsass/src/constants.cpp
compiling ./libsass/src/context.cpp
compiling ./libsass/src/cssize.cpp
compiling ./libsass/src/emitter.cpp
compiling ./libsass/src/environment.cpp
compiling ./libsass/src/error_handling.cpp
compiling ./libsass/src/eval.cpp
compiling ./libsass/src/eval_selectors.cpp
compiling ./libsass/src/expand.cpp
compiling ./libsass/src/extender.cpp
compiling ./libsass/src/extension.cpp
compiling ./libsass/src/file.cpp
compiling ./libsass/src/fn_colors.cpp
compiling ./libsass/src/fn_lists.cpp
compiling ./libsass/src/fn_maps.cpp
compiling ./libsass/src/fn_miscs.cpp
compiling ./libsass/src/fn_numbers.cpp
compiling ./libsass/src/fn_selectors.cpp
compiling ./libsass/src/fn_strings.cpp
compiling ./libsass/src/fn_utils.cpp
compiling ./libsass/src/inspect.cpp
compiling ./libsass/src/json.cpp
compiling ./libsass/src/lexer.cpp
compiling ./libsass/src/listize.cpp
compiling ./libsass/src/operators.cpp
compiling ./libsass/src/output.cpp
compiling ./libsass/src/parser.cpp
compiling ./libsass/src/parser_selectors.cpp
compiling ./libsass/src/plugins.cpp
compiling ./libsass/src/position.cpp
compiling ./libsass/src/prelexer.cpp
compiling ./libsass/src/remove_placeholders.cpp
compiling ./libsass/src/sass.cpp
compiling ./libsass/src/sass2scss.cpp
compiling ./libsass/src/sass_context.cpp
compiling ./libsass/src/sass_functions.cpp
compiling ./libsass/src/sass_values.cpp
compiling ./libsass/src/source.cpp
compiling ./libsass/src/source_map.cpp
compiling ./libsass/src/stylesheet.cpp
compiling ./libsass/src/to_value.cpp
compiling ./libsass/src/units.cpp
compiling ./libsass/src/utf8_string.cpp
compiling ./libsass/src/util.cpp
compiling ./libsass/src/util_string.cpp
compiling ./libsass/src/values.cpp
linking shared-object sassc/libsass.dll
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/14.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot find allocator.o: No such file or directory
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/14.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot find shared_ptr.o: No such file or directory
collect2.exe: error: ld returned 1 exit status
make: *** [Makefile:221: libsass.dll] Error 1

make failed, exit code 2

Gem files will remain installed in D:/jruby-head/lib/ruby/gems/shared/gems/sassc-2.4.0 for inspection.
Results logged to D:/jruby-head/lib/ruby/gems/shared/extensions/universal-java-1.8/3.1.0/sassc-2.4.0/gem_make.out
Error: Process completed with exit code 1.
```

--- a/core/src/main/java/org/jruby/util/Dir.java
+++ b/core/src/main/java/org/jruby/util/Dir.java
@@ -976,7 +976,7 @@ public class Dir {
                             buf.append(base);
                             buf.append( isRoot(base) ? EMPTY : SLASH );
                             buf.append( getBytesInUTF8(file) );
-                            resource = JRubyFile.createResource(runtime, cwd, RubyString.byteListToString(buf));
+                            resource = JRubyFile.createResource(runtime, cwd, new String(buf.unsafeBytes(), buf.begin(), buf.length(), enc.getCharset()));
                             boolean dirMatch = SLASH_INDEX == end-1 && resource.isDirectory();
                             if ( dirMatch || SLASH_INDEX == -1 ) {
                                 if (scheme != null) {

--- a/core/src/main/java/org/jruby/util/Dir.java
+++ b/core/src/main/java/org/jruby/util/Dir.java
@@ -976,8 +976,11 @@ public class Dir {
                             buf.append(base);
                             buf.append( isRoot(base) ? EMPTY : SLASH );
                             buf.append( getBytesInUTF8(file) );
-                            resource = JRubyFile.createResource(runtime, cwd, new String(buf.unsafeBytes(), buf.begin(), buf.length(), enc.getCharset()));
-                            boolean dirMatch = SLASH_INDEX == end-1 && resource.isDirectory();
+                            boolean dirMatch = false;
+                            if (SLASH_INDEX == end - 1) {
+                                resource = JRubyFile.createResource(runtime, cwd, new String(buf.unsafeBytes(), buf.begin(), buf.length(), enc.getCharset()));
+                                dirMatch = resource.isDirectory();
+                            }
                             if ( dirMatch || SLASH_INDEX == -1 ) {
                                 if (scheme != null) {
                                     byte[] bufBytes = buf.bytes();
@@ -990,7 +993,7 @@ public class Dir {
                                 if ( status != 0 ) break;
                                 continue;
                             }
-                            if (resource.isDirectory()) links.add(new DirGlobber(buf));
+                            links.add(new DirGlobber(buf));
                             buf = new ByteList(20);
                             buf.setEncoding(enc);
                         }
@@ -1001,11 +1004,14 @@ public class Dir {
                     for ( DirGlobber globber : links ) {
                         final ByteList link = globber.link;
                         if ( status == 0 ) {
-                            final int len = link.getRealSize();
-                            buf.length(0);
-                            buf.append(link);
-                            buf.append(path, SLASH_INDEX, end - SLASH_INDEX);
-                            status = glob_helper(runtime, cwd, scheme, buf, buf.getBegin() + len, flags, func, arg);
+                            resource = JRubyFile.createResource(runtime, cwd, new String(link.unsafeBytes(), link.begin(), link.length(), enc.getCharset()));
+                            if ( resource.isDirectory() ) {
+                                final int len = link.getRealSize();
+                                buf.length(0);
+                                buf.append(link);
+                                buf.append(path, SLASH_INDEX, end - SLASH_INDEX);
+                                status = glob_helper(runtime, cwd, scheme, buf, buf.getBegin() + len, flags, func, arg);
+                            }
                         }
                     }
                     break mainLoop;

--- a/core/src/main/java/org/jruby/util/Dir.java
+++ b/core/src/main/java/org/jruby/util/Dir.java
@@ -973,16 +973,24 @@ public class Dir {
                         }
                         if ( fnmatch(magic, 0, magic.length, fileBytes, 0, fileBytes.length, flags) == 0 ) {
                             buf.length(0);
-                            if (scheme != null && SLASH_INDEX == -1) buf.append(scheme);
                             buf.append(base);
                             buf.append( isRoot(base) ? EMPTY : SLASH );
                             buf.append( getBytesInUTF8(file) );
-                            if ( SLASH_INDEX == -1 ) {
+                            resource = JRubyFile.createResource(runtime, cwd, RubyString.byteListToString(buf));
+                            boolean dirMatch = SLASH_INDEX == end-1 && resource.isDirectory();
+                            if ( dirMatch || SLASH_INDEX == -1 ) {
+                                if (scheme != null) {
+                                    byte[] bufBytes = buf.bytes();
+                                    buf.length(0);
+                                    buf.append(scheme);
+                                    buf.append(bufBytes);
+                                }
+                                if (dirMatch) buf.append(SLASH);
                                 status = func.call(buf.getUnsafeBytes(), 0, buf.getRealSize(), enc, arg);
                                 if ( status != 0 ) break;
                                 continue;
                             }
-                            links.add(new DirGlobber(buf));
+                            if (resource.isDirectory()) links.add(new DirGlobber(buf));
                             buf = new ByteList(20);
                             buf.setEncoding(enc);
                         }
@@ -993,14 +1001,11 @@ public class Dir {
                     for ( DirGlobber globber : links ) {
                         final ByteList link = globber.link;
                         if ( status == 0 ) {
-                            resource = JRubyFile.createResource(runtime, cwd, RubyString.byteListToString(link));
-                            if ( resource.isDirectory() ) {
-                                final int len = link.getRealSize();
-                                buf.length(0);
-                                buf.append(link);
-                                buf.append(path, SLASH_INDEX, end - SLASH_INDEX);
-                                status = glob_helper(runtime, cwd, scheme, buf, buf.getBegin() + len, flags, func, arg);
-                            }
+                            final int len = link.getRealSize();
+                            buf.length(0);
+                            buf.append(link);
+                            buf.append(path, SLASH_INDEX, end - SLASH_INDEX);
+                            status = glob_helper(runtime, cwd, scheme, buf, buf.getBegin() + len, flags, func, arg);
                         }
                     }
                     break mainLoop;

--- a/test/jruby/test_dir.rb
+++ b/test/jruby/test_dir.rb
@@ -213,6 +213,18 @@ class TestDir < Test::Unit::TestCase
     FileUtils.rm_r("testDir_3") rescue nil
   end
 
+  def test_glob_dir
+    Dir.mkdir('testDir_4')
+    Dir.mkdir('testDir_4/a')
+    Dir.mkdir('testDir_4/b')
+    FileUtils.touch('testDir_4/file.txt')
+    assert_equal(Dir.glob('testDir_4/*/'), ['testDir_4/a/', 'testDir_4/b/'])
+    Dir.chdir("testDir_4") do
+      assert_equal(Dir.glob('*/'), ['a/', 'b/'])
+    end
+    FileUtils.rm_r("testDir_4") rescue nil
+  end
+
   def test_dir_dot_does_not_throw_exception
     # just makes sure this doesn't throw a Java exception
     Dir['.']

